archunit/src/main/java/com/tngtech/archunit/core/domain/JavaCall.java

```diff
@@ -15,17 +15,38 @@
  */
 package com.tngtech.archunit.core.domain;
 
+import com.google.common.base.Function;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Sets;
 import com.tngtech.archunit.PublicAPI;
 import com.tngtech.archunit.base.DescribedPredicate;
 import com.tngtech.archunit.core.domain.AccessTarget.CodeUnitCallTarget;
 import com.tngtech.archunit.core.domain.JavaAccess.Predicates.TargetPredicate;
+import com.tngtech.archunit.core.importer.CaughtThrowable;
+import com.tngtech.archunit.core.importer.DomainBuilders;
 import com.tngtech.archunit.core.importer.DomainBuilders.JavaAccessBuilder;
 
+import java.util.Set;
+
 import static com.tngtech.archunit.PublicAPI.Usage.ACCESS;
 
 public abstract class JavaCall<T extends CodeUnitCallTarget> extends JavaCodeUnitAccess<T> {
-    JavaCall(JavaAccessBuilder<T, ?> builder) {
+    private static final Function<JavaClassDescriptor, CaughtThrowable> TO_CAUGHT_THROWABLE_CONVERTER = new Function<JavaClassDescriptor, CaughtThrowable>() {
+
+        @Override
+        public CaughtThrowable apply(JavaClassDescriptor input) {
+            return new CaughtThrowable(input);
```

Comment:
In this minimal version, only the exception type is retained in `CaughtThrowable`

This is a pretty large PR as it is, I'd suggest that any enhancements be added in a separate PR